compaction/compaction_manager.cc

```diff
@@ -622,10 +622,11 @@ sstables::compaction_stopped_exception compaction_manager::task::make_compaction
     return sstables::compaction_stopped_exception(s->ks_name(), s->cf_name(), _compaction_data.stop_requested);
 }
 
-compaction_manager::compaction_manager(config cfg, abort_source& as)
-    : _compaction_sg(cfg.compaction_sched_group)
-    , _maintenance_sg(cfg.maintenance_sched_group)
-    , _compaction_controller(make_compaction_controller(_compaction_sg, cfg.static_shares, [this, available_memory = cfg.available_memory] () -> float {
+compaction_manager::compaction_manager(std::function<config()> get_cfg, abort_source& as)
```

Comment:
Config getter :scream: 
I guess it's because of #7316 ?

Reply:
Yes, exactly.

Reply:
Awesome! The a246b6d3eb0de2bb359b32a269b7247d20ca9fbe suffers from the same, so likely the issue is going to be fixed really soon :)

Reply:
Ouch